geo/src/main/java/org/apache/carbondata/geo/GeoHashImpl.java

```diff
@@ -38,10 +41,66 @@
  * columns.
  */
 public class GeoHashImpl extends CustomIndex<List<Long[]>> {
+  private static final Logger LOGGER =
+      LogServiceFactory.getLogService(GeoHashImpl.class.getName());
+
+  // conversion factor of angle to radian
+  private static final double CONVERT_FACTOR = 180.0;
+  // Earth radius
+  private static final double EARTH_RADIUS = 6371004.0;
+  // Longitude of coordinate origin
+  // private double oriLongitude = 0;
+  // Latitude of coordinate origin
+  private double oriLatitude;
+  // User defined maximum longitude of map
+  private double userDefineMaxLongitude;
+  // User defined maximum latitude of map
+  private double userDefineMaxLatitude;
+  // User defined map minimum longitude
+  private double userDefineMinLongitude;
+  // User defined map minimum latitude
+  private double userDefineMinLatitude;
+  // The maximum longitude of the completed map after calculation
+  private double CalculateMaxLongitude;
+  // The maximum latitude of the completed map after calculation
+  private double CalculateMaxLatitude;
+  // Grid length is in meters
+  private int gridSize;
+  // cos value of latitude of origin of coordinate
+  private double mCos;
+  // The degree of Y axis corresponding to each grid size length
+  private double deltaY;
+  // Each grid size length should be the degree of X axis
+  private double deltaX;
+  // Degree * coefficient of Y axis corresponding to each grid size length
+  private double deltaYByRatio;
+  // Each grid size length should be X-axis Degree * coefficient
+  private double deltaXByRatio;
+  // The number of knives cut for the whole area (one horizontally and one vertically)
+  // is the depth of quad tree
+  private int cutLevel;
+  // used to convert the latitude and longitude of double type to int type for calculation
+  private int conversionRatio;
+  // * Constant of coefficient
+  private double lon0ByRation;
+  // * Constant of coefficient
+  private double lat0ByRation;
+
+
   /**
    * Initialize the geohash index handler instance.
-   * @param handlerName
-   * @param properties
+   * the properties is like that:
+   * TBLPROPERTIES ('INDEX_HANDLER'='mygeohash',
+   * 'INDEX_HANDLER.mygeohash.type'='geohash',
+   * 'INDEX_HANDLER.mygeohash.sourcecolumns'='longitude, latitude',
+   * 'INDEX_HANDLER.mygeohash.gridSize'='50'
+   * 'INDEX_HANDLER.mygeohash.minLongitude'='0'
+   * 'INDEX_HANDLER.mygeohash.maxLongitude'='0'
+   * 'INDEX_HANDLER.mygeohash.minLatitude'='100'
+   * 'INDEX_HANDLER.mygeohash.maxLatitude'='100'
+   * 'INDEX_HANDLER.mygeohash.orilatitude''8')
```

Comment:
this GeoHashImpl.java cannot be the default implementation as  it depends on grid size, min max latitude longitude and original latitude longitude. 

will it work if these parameters set to 0 also ? then it can be default implementation. If it cannot work we can have it as a custom implementation and introduce simple default implementation that can work on any geo spatial polygon query. (with float longitude, latitude)

Reply:
that is just an example how to configure the parameters, has change the comment.

Reply:
No, I am not talking about comment. I am talking about whole file. This file cannot be the default implementation